### ISSUE TITLE
I can't compile

### ISSUE THREAD

**Issue by Mahoyomu** [minetest/minetest#14773]
After trying for about a whole week, checking out over 10 possible solution posts, I end up failing compile minetest again. And I can't take it any more so I open a issue to ask for help.
I use VS2022, completely follow steps in the tutorial video on youtube, and installed every environment I saw in the posts, thought may be what caused this, but none of those solutions worked.
Cmake gui just keeping reporting error like "CMake CXX Compiler Not Found" or "is not able to compile simple test program", no matter the path I gave to Cmake is MinGW or "bin/Hostx64/x64/cl.exe" of VS2022.
Please help me with it. I can't test anything if I can't compile minetest.

**Comment by Mahoyomu** [comment]
I use Win10 x64 btw

**Comment by DragonWrangler1** [comment]
Here is a link to the actual mine test documentation. Incan the YouTube video left something out.

https://github.com/minetest/minetest/blob/master/doc/compiling/windows.md

**Comment by SmallJoker** [comment]
There's a specific script that sets up the paths necessary for the compiler. You should find a link in your start menu called "x64 Native Tools Command Prompt for VS 2022" (or similar). Have you yet tried to run `cmake` (or `cmake-gui`) inside such a terminal instance?

**Comment by Mahoyomu** [comment]
> Here is a link to the actual minetest documentation. Incase the YouTube video left something out.
> 
> https://github.com/minetest/minetest/blob/master/doc/compiling/windows.md
> 
> unfortunantly I can’t help much since I use Linux.

I first followed instrustions in this documentation, and it didn't work so I went searching and found the Youtube video. Both didn't work for me

**Comment by Mahoyomu** [comment]
> There's a specific script that sets up the paths necessary for the compiler. You should find a link in your start menu called "x64 Native Tools Command Prompt for VS 2022" (or similar). Have you yet tried to run (or ) inside such a terminal instance?`cmake``cmake-gui`

No, but I just tried this and it reported an error said `The system cannot find the file specified.`

**Comment by Mahoyomu** [comment]
I just figured out how to solve those problems above and solved them, but now `VS2022` has `warning C4819` `error C2001` `error C2065`, and I can't compile again, I don't know what to do with it
